libexec/alluxio-config.sh

```diff
@@ -86,6 +85,26 @@ ALLUXIO_JAVA_OPTS+=" -Dlog4j.configuration=file:${ALLUXIO_CONF_DIR}/log4j.proper
 ALLUXIO_JAVA_OPTS+=" -Dorg.apache.jasper.compiler.disablejsr199=true"
 ALLUXIO_JAVA_OPTS+=" -Djava.net.preferIPv4Stack=true"
 
+ALLUXIO_CLIENT_CLASSPATH="${ALLUXIO_CONF_DIR}/:${ALLUXIO_CLASSPATH}:${ALLUXIO_ASSEMBLY_CLIENT_JAR}"
+ALLUXIO_SERVER_CLASSPATH="${ALLUXIO_CONF_DIR}/:${ALLUXIO_CLASSPATH}:${ALLUXIO_ASSEMBLY_SERVER_JAR}"
+
+####################################################################################################
+## Start reading site-properties to set certain variables.
+####################################################################################################
+function getConf {
+  CLASS="alluxio.cli.GetConf"
+  ALLUXIO_SHELL_JAVA_OPTS+=" -Dalluxio.logger.type=Console"
+  "${JAVA}" -cp ${ALLUXIO_CLIENT_CLASSPATH} ${ALLUXIO_SHELL_JAVA_OPTS} ${CLASS} "$1"
+}
+
+if [[ -z "${ALLUXIO_LOGS_DIR}" ]]; then
+    ALLUXIO_LOGS_DIR=$(getConf "alluxio.logs.dir")
+    ALLUXIO_JAVA_OPTS+=" -Dalluxio.logs.dir=${ALLUXIO_LOGS_DIR}"
+fi
+####################################################################################################
+## End reading site-properties
+####################################################################################################
+
```

Comment:
We could. But putting these lines here will cause the less change. If we put these to the bottom of this file, then we need to modify these several XXX_JAVA_OPTS again.